src/clients/dispatcher/action-listener.ts

```diff
@@ -18,7 +23,23 @@ enum ListenStrategy {
   LISTEN_STRATEGY_V2 = 2,
 }
 
-export interface Action extends AssignedAction {}
+export const ActionObject = z.object({
```

Comment:
we're using zod here because we need to validate the payload?
i recently swapped this to extends assigned actions so generating keeps this in sync

Reply:
yeah for the webhook handler, it's not necessary because the signature is checked, but I figured it's good to ensure the data is fully valid

Reply:
I guess it's fine